sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/arithmetic.scala

```diff
@@ -314,6 +314,27 @@ case class Divide(left: Expression, right: Expression) extends DivModLike {
   override def evalOperation(left: Any, right: Any): Any = div(left, right)
 }
 
+@ExpressionDescription(
+  usage = "expr1 _FUNC_ expr2 - Returns `expr1`/`expr2`. It performs integral division.",
+  examples = """
+    Examples:
+      > SELECT 3 _FUNC_ 2;
+       1
+  """,
+  since = "3.0.0")
+case class IntegralDivide(left: Expression, right: Expression) extends DivModLike {
+
+  override def inputType: AbstractDataType = IntegralType
+
+  override def symbol: String = "/"
+  override def sqlOperator: String = "div"
+
+  private lazy val div: (Any, Any) => Any = dataType match {
+    case i: IntegralType => i.integral.asInstanceOf[Integral[Any]].quot
+  }
+  override def evalOperation(left: Any, right: Any): Any = div(left, right)
```

Comment:
Sorry I may not recall it very clearly. Can you check Hive and other databases and see if the result type of `div` is always long?

Reply:
Sure, so: 
- Hive returns always long;
 - Postgres and SQLServer don't have a `div` operator but they perform integral division when the operands are integrals and return the datatype of the operands (eg. `select 3 / 2` returns an integer);
 - Oracle doesn't support it.

So the behavior is not homogeneous among the RDBMs

Reply:
Then I'd prefer always returning long, since it was the behavior before. We can consider changing the behavior in another PR.

Reply:
+1 for @cloud-fan 's suggestion.

Reply:
Yeah, I think it is reasonable as that is what we defined: `Hive Long Division: 'DIV'` in `AstBuilder.scala`.